### PR TITLE
add example for ease functions

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Followup to #15675 
- Add an example showcasing the functions

## Solution

- Add an example showcasing the functions
- Some of the functions from the interpolation crate are messed up, fixed in #15706

ease

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
*Prime* release note eye candy here :D

File: examples/animation/ease_functions.rs [hunk 1]
```diff
@@ -0,0 +1,178 @@
+//! Demonstrates the builtin easing functions.
```
Comment (alice-i-cecile) [inline examples/animation/ease_functions.rs:1]:
```suggestion
//! Demonstrates the behavior of the built-in easing functions.
```

**Review by alice-i-cecile — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -1280,6 +1280,17 @@ description = "Bezier curve example showing a cube following a cubic curve"
...
 wasm = true
 
+[[example]]
+name = "ease_functions"
+path = "examples/animation/ease_functions.rs"
+doc-scrape-examples = true
+
+[package.metadata.example.ease_functions]
+name = "Ease Functions"
+description = "Showcase of the built-in ease functions"
```
Comment (Jondolf) [inline Cargo.toml:1290]: Nit: "Easing functions" might read better? The example also says uses it. Could consider renaming the file to match as well, but no strong opinion either way
```suggestion
description = "Showcases the built-in easing functions"
```

File: examples/README.md [hunk 3]
```diff
@@ -200,6 +200,7 @@ Example | Description
 [Color animation](../examples/animation/color_animation.rs) | Demonstrates how to animate colors using mixing and splines in different color spaces
 [Cubic Curve](../examples/animation/cubic_curve.rs) | Bezier curve example showing a cube following a cubic curve
 [Custom Skinned Mesh](../examples/animation/custom_skinned_mesh.rs) | Skinned mesh example with mesh and joints data defined in code
+[Ease Functions](../examples/animation/ease_functions.rs) | Showcase of the built-in ease functions
```
Comment (Jondolf) [inline examples/README.md:203]:
```suggestion
[Easing Functions](../examples/animation/ease_functions.rs) | Showcases the built-in easing functions
```

File: examples/animation/ease_functions.rs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        // Draw the curve
+        let f = easing::EasingCurve::ease(*function);
+        gizmos.linestrip_2d(
+            (0..(samples + 1)).map(|i| {
+                Vec2::new(
+                    i as f32 / samples as f32 * size + transform.translation.x,
+                    f.sample(i as f32 / samples as f32).unwrap() * size
+                        + transform.translation.y
+                        + 15.0,
+                )
```
Comment (Jondolf) [inline examples/animation/ease_functions.rs:161]: Maybe the sampled value could be split on another line and in a variable, possibly with a comment?

**Review by Jondolf — APPROVED** [review]
LGTM! Just a few stylistic nits, not blocking.

**Comment by alice-i-cecile** [comment]
Merge this at your leisure @mockersf  :)

File: Cargo.toml [hunk 5]
```diff
@@ -1280,6 +1280,17 @@ description = "Bezier curve example showing a cube following a cubic curve"
 category = "Animation"
 wasm = true
 
+[[example]]
+name = "ease_functions"
+path = "examples/animation/ease_functions.rs"
+doc-scrape-examples = true
+
+[package.metadata.example.ease_functions]
+name = "Ease Functions"
```
Comment (Jondolf) [inline Cargo.toml:1289]: Whoops missed this one in my suggestions (in case you decide to incoporate them)
```suggestion
name = "Easing Functions"
```

**Review by tychedelia — APPROVED** [review]
awesome, these are really cool!

**Comment by mweatherley** [comment]
I didn't realize `SineIn` and `SineOut` went so far out of bounds; that's crazy!

**Pull request merged**